photon-apple/build.gradle

```diff
@@ -0,0 +1,135 @@
+plugins {
+    id 'java-library'
+}
+
+def isMacOS = System.getProperty("os.name").toLowerCase().contains("mac")
+def forceStubs = project.hasProperty('photonAppleForceStubs')
+def useRealImplementation = isMacOS && !forceStubs
+
+// Configure Java version based on build mode
```

Comment:
Can we document somewhere the reason driving all this trickery around java versions? I know some of this is in Discord, but would like this captured